src/coreclr/jit/lsrabuild.cpp

```diff
@@ -1508,6 +1528,10 @@ void LinearScan::buildUpperVectorSaveRefPositions(GenTree* tree, LsraLocation cu
                     newRefPosition(upperVectorInterval, currentLoc, RefTypeUpperVectorSave, tree, RBM_FLT_CALLEE_SAVED);
                 varInterval->isPartiallySpilled = true;
                 pos->skipSaveRestore            = blockAlwaysReturn;
```

Comment:
This is preexisting, but why is it ok to ignore save/restore in a block that always returns? How does that work if there are more uses of the vector local after the call?

Reply:
yes, I think there was an example I was hitting this, but may be because I was not handling the upper vector save/restore correctly at other places. So this should not be needed. I will remove it.

Reply:
Ah, I think I understand now... `skipSaveRestore` gets reset again by `buildUpperVectorRestoreRefPosition` if we end up seeing a use. So I think having `BBJ_RETURN` in the check there would be ok.